lib/services/ngsi/ngsiService.js

```diff
@@ -690,7 +690,19 @@ function sendQueryValue(entityName, attributes, typeInformation, token, callback
  */
 
 function updateTrust(deviceGroup, deviceInformation, trust, response, callback) {
-    callback(null, response);
+    if (deviceGroup && response && response.body) {
```

Comment:
it could be at the level of the configuration of the agent e.g. `authenticationo.oauth_offline_token   = 1`

Reply:
I implemented this because in my tests, each time I requested a new `access_token` to Keyrock using the `refresh_token`, the latter one became invalid. I will also ask this to Keyrock team.

Reply:
After checking the standard it says:

>    The authorization server MAY issue a new refresh token, in which case
   the client MUST discard the old refresh token and replace it with the
   new refresh token.  The authorization server MAY revoke the old
   refresh token after issuing a new refresh token to the client.  If a
   new refresh token is issued, the refresh token scope MUST be
   identical to that of the refresh token included by the client in the
   request.

So I understand that in this case, Keyrock implementation is right.

Reply:
hi @dcalvoalonso as mentioned in the email with upm team, this is the case if you use a "normal" refresh token and not the one obtained with 'offline_access' [scope] [1]. why it is better to use the 'offline_access' modality (i.e. a never expiring refresh token)?
in case, for whatever reason (no update for the sensor over a long time, broken connectivity, iot agent was down for some time), you don't refresh the refresh_token before its expiration, you would need to do again a manual authentication and update the refresh token in the configuration. 

[1]: https://openid.net/specs/openid-connect-core-1_0.html#OfflineAccess.

Reply:
see here some discussion on refresh token expiration: https://medium.com/@bantic/more-oauth-2-0-surprises-the-refresh-token-1831d71f4af6

Reply:
Fixed with 2aab653c5bdcf5703ae470cd81925f7289ae4e45